src/formatDate.js

```diff
@@ -8,7 +8,42 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const separator = fromFormat.pop();
```

Comment:
Using `Array.pop()` here is mutating the input `fromFormat` array, which can lead to unexpected side-effects. Consider using a different approach that does not modify the input array.